packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -1630,6 +1642,24 @@ contract OPContractsManagerInteropMigrator is OPContractsManagerBase {
             newDisputeGameFactory.setImplementation(GameTypes.SUPER_CANNON, IDisputeGame(address(newSuperFDG)));
             newDisputeGameFactory.setInitBond(GameTypes.SUPER_CANNON, _input.gameParameters.initBond);
         }
+
+        // Safety assertions, check that all systems use the same contracts.
+        for (uint256 i = 0; i < portals.length; i++) {
+            // Check that the ETHLockbox is the same.
+            if (portals[i].ethLockbox() != newEthLockbox) {
+                revert OPContractsManagerInteropMigrator_ETHLockboxMismatch();
+            }
+
+            // Check that the DisputeGameFactory is the same.
+            if (portals[i].disputeGameFactory() != newDisputeGameFactory) {
+                revert OPContractsManagerInteropMigrator_DisputeGameFactoryMismatch();
+            }
+
+            // Check that the AnchorStateRegistry is the same.
+            if (portals[i].anchorStateRegistry() != newAnchorStateRegistry) {
+                revert OPContractsManagerInteropMigrator_AnchorStateRegistryMismatch();
+            }
```

Comment:
We should check that the anchor state registry's disputeGameFactory is the one we expect (unless that check is somewhere else already).

um, also, AnchorStateRegistry has a reference to SystemConfig:
https://github.com/ethereum-optimism/optimism/blob/52100a6a0020a97ac4de51fff33545238996029d/packages/contracts-bedrock/src/dispute/AnchorStateRegistry.sol#L34-L35

That doesn't work when it's shared across multiple chains. Am I missing something?

Reply:
Oh yeah, this is awkward. Though all systemConfigs in an interop set will behave the same for checking guardian and pausability - which are the only two functionalities we need in the ASR.
This should be replaced with the `superchainConfig` and `ethlockbox` to do the same thing.